src/agent/debugger/src/stack.rs

```diff
@@ -14,31 +14,63 @@ use serde::{Serialize, Serializer};
 use win_util::memory;
 use winapi::{shared::minwindef::DWORD, um::winnt::HANDLE};
 
-use crate::dbghelp::{self, DebugHelpGuard, ModuleInfo};
+use crate::dbghelp::{self, DebugHelpGuard, ModuleInfo, SymLineInfo};
 
 const UNKNOWN_MODULE: &str = "<UnknownModule>";
 
+/// The file and line number for frame in the calls stack.
 #[derive(Clone, Debug, Hash, PartialEq)]
-pub enum DebugFunctionLocation {
+pub struct FileInfo {
+    pub file: String,
+    pub line: u32,
+}
+
+impl From<&SymLineInfo> for FileInfo {
+    fn from(sym_line_info: &SymLineInfo) -> Self {
+        let file = sym_line_info.filename().to_string_lossy().into();
+        let line = sym_line_info.line_number();
+        FileInfo { file, line }
+    }
+}
+
+#[derive(Clone, Debug, Hash, PartialEq)]
+pub struct DebugFunctionLocation {
     /// File/line if available
     ///
     /// Should be stable - ASLR and JIT should not change source position,
     /// but some precision is lost.
     ///
     /// We mitigate this loss of precision by collecting multiple samples
     /// for the same hash bucket.
-    Line { file: String, line: u32 },
+    pub file_info: Option<FileInfo>,
 
     /// Offset if line information not available.
-    Offset { disp: u64 },
+    pub displacement: u64,
+}
+
+impl DebugFunctionLocation {
+    pub fn new(displacement: u64) -> Self {
+        DebugFunctionLocation {
+            displacement,
+            file_info: None,
+        }
+    }
+
+    pub fn new_with_file_info(displacement: u64, file_info: FileInfo) -> Self {
+        DebugFunctionLocation {
+            displacement,
+            file_info: Some(file_info),
+        }
+    }
 }
 
 impl Display for DebugFunctionLocation {
     fn fmt(&self, formatter: &mut Formatter) -> fmt::Result {
-        match self {
-            DebugFunctionLocation::Line { file, line } => write!(formatter, "{}:{}", file, line)?,
-            DebugFunctionLocation::Offset { disp } => write!(formatter, "0x{:x}", disp)?,
-        };
+        if let Some(file_info) = &self.file_info {
+            write!(formatter, "{}:{}", file_info.file, file_info.line)?;
```

Comment:
We could impl `Display` for `FileInfo`, then use that impl here and in the impl for `DebugStackFrame`.